packages/gatsby-image/src/index.js

```diff
@@ -168,10 +154,16 @@ class Image extends React.Component {
     this.handleRef = this.handleRef.bind(this)
   }
 
+  componentDidMount() {
+    if (this.props.critical) {
+      this.setState({imgLoaded: true});
+    }
+  }
+
   handleRef(ref) {
     if (this.state.IOSupported && ref) {
       listenToIntersections(ref, () => {
-        this.setState({ isVisible: true, imgLoaded: false })
+        this.setState({ isVisible: true })
```

Comment:
Not sure why this was set to false here.  Reseting the state to false causes issues with critical images, and I could not find where non-critical images required this state change.

Reply:
hmmm yeah, don't remember either 😅